edit/edit.js

```diff
@@ -52,10 +57,17 @@ msg.onExtension(request => {
   const {style} = request;
   switch (request.method) {
     case 'styleUpdated':
-      if (editor.style.id === style.id &&
-        !['editPreview', 'editPreviewEnd', 'editSave', 'config'].includes(request.reason)) {
-        Promise.resolve(request.codeIsUpdated === false ? style : API.styles.get(style.id))
-          .then(newStyle => editor.replaceStyle(newStyle, request.codeIsUpdated));
+      if (editor.style.id === style.id) {
+        if (!['editPreview', 'editPreviewEnd', 'editSave', 'config'].includes(request.reason)) {
+          Promise.resolve(request.codeIsUpdated === false ? style : API.styles.get(style.id))
+            .then(newStyle => {
+              editor.replaceStyle(newStyle, request.codeIsUpdated);
+
+              if (['success-linking', 'success-revoke'].includes(request.reason)) {
+                updateUI(newStyle);
```

Comment:
As the `replaceStyle` is a `async` function and I don't like to add an async into this anonymous function so we just pass the newStyle into `updateUI`.